types/document.d.ts

```diff
@@ -225,7 +225,7 @@ declare module 'mongoose' {
     toJSON<T = DocType>(options: ToObjectOptions & { flattenMaps: false }): LeanDocument<T>;
 
     /** Converts this document into a plain-old JavaScript object ([POJO](https://masteringjs.io/tutorials/fundamentals/pojo)). */
-    toObject<T = DocType>(options?: ToObjectOptions): LeanDocument<T>;
+    toObject<T = DocType>(options?: ToObjectOptions): Require_id<LeanDocument<T>>;
```

Comment:
In the interest of backwards compat, why not make this just `toObject<T = LeanDocument<DocType>>` and return `Require_id<T>`? That should allow people to bypass going through `LeanDocument`, which seems to be causing part of the problem.

Reply:
Definitely an option; by doing that you will not be converting any mongoose.Types.Array types to real javascript arrays and any embedded subdocuments won't get converted to POJOs.